doc/zk-integration.md

```diff
@@ -0,0 +1,217 @@
+# Integration with ZK rollups
+
+Similarly to their optimistic alternatives, ZK rollups need to instantiate their derivation pipeline in order to
+integrate with Espresso. There are a number of alternatives for such integration, depending on specific constraints the
+rollup may have such as using a particular data availability layer in addition to Tiramisu, or the need to update their
+state on the L1 at a faster pace than the Espresso light client contract does.
+
+In the following, we assume that the rollup relies on Tiramisu for data availability and explore two different
+integrations flavors. With integration 1, the rollup relies on the Espresso light client contract to fetch the latest
+Espresso state updates. while with integration 2 the rollup checks Espresso consensus directly inside its circuit.
+
+For both integrations:
+
+- We describe the high level structure of the circuit as well as the rollup L1 contract and how they are related.
+- We describe the use of an Espresso liveness escape hatch in the rollup L1 contract.
+
+When a rollup is not integrated with Espresso, i.e. it uses its own centralized sequencer, its circuit only needs to
+check the correct update of the zkVM as depicted in Figure 1. This is also the circuit used when the escape hatch is
+activated, as in this case the rollup falls back to using its own centralized sequencer. When the escape hatch is not
```

Comment:
The introduction has been improved in 4c63c5e6.

> can we lay down more background? I find it a bit ambiguous. Clarifying the terms used is also helpful. Identifying the major roles/components and add an overall flow at the beginning would be nice.

Added a reference to the main README.md that introduces term and has a detailed sequence diagram. Also referenced this document from the README.md

> "integrate with Espresso" means the marketplace or finality gadget? as one can use its own sequencer and still use the finality gadget.

Finality gadget. This should be less ambiguous now.

> maybe clarify that "escape hatch" means a liveness issue at the finality gadget level?

A sentence has been added regarding what the escape hatch and how it relates to the finality gadget.

> also at this level, centralized sequencer here is kind of out-of-the-blue, do we want to say "backup sequencer" instead? Because in the marketplace, there's always a winner that's a centralized party always, right?

Yes, the term *backup sequencer* is better.